### PR TITLE
wave3: Huber Loss for Robust Regression (cross-dataset)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Replace MSE loss with Huber loss (smooth L1) for more robust regression training. MSE penalizes outliers quadratically, which can destabilize training when CFD fields contain rare high-magnitude anomalies (e.g. boundary layer separation, shock-like gradients near sharp geometric features). Huber loss behaves like L1 for large residuals (linear penalty) and L2 for small residuals (smooth near zero), controlled by delta. This may improve generalization by reducing the model's sensitivity to hard-to-predict outlier points.

**Expected outcome:** Improved surface pressure MAE on TandemFoil (target: <26.06), surface MSE on AirfRANS (target: <0.000627), surface_rel_l2_pct on DrivAerML (target: <3.997%), and field_mse on TandemFoil Paper.

## Implementation

Add `--loss-fn huber --huber-delta DELTA` flags to `train.py`. Wrap the primary loss computation:

```python
# In train.py, loss computation block:
import torch.nn.functional as F

if args.loss_fn == 'huber':
    # Huber loss: L2 when |error| < delta, L1 scaled otherwise
    loss = F.huber_loss(pred, target, reduction='mean', delta=args.huber_delta)
elif args.loss_fn == 'mse':
    loss = F.mse_loss(pred, target, reduction='mean')
else:
    loss = F.mse_loss(pred, target, reduction='mean')
```

Add to argparse:
```python
parser.add_argument('--loss-fn', type=str, default='mse', choices=['mse', 'huber'],
                    help='Loss function: mse (default) or huber (smooth L1)')
parser.add_argument('--huber-delta', type=float, default=1.0,
                    help='Delta threshold for Huber loss (default: 1.0)')
```

**Key insight on delta selection:** Since surface pressure values are normalized, delta should be tuned per-dataset. Start with delta=1.0 (standard Huber). Also test delta=0.1 (more L1-like, stronger outlier robustness) and delta=0.5. For DrivAerML's surface_rel_l2_pct metric (already L2-based), delta=1.0 is likely best.

### TandemFoil (golden config + Huber loss)
```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil \
  --optimizer lion --lr 1e-4 --cosine-t-max 10 \
  --grad-clip 1.0 --weight-decay 1e-2 --no-use-ema \
  --model-slices 64 --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --enable-fourier --enable-te-coord-frame --enable-cp-panel \
  --enable-cp-panel-tandem-only --asinh-pressure \
  --residual-prediction --enable-pressure-prior-addition \
  --loss-fn huber --huber-delta 1.0 \
  --epochs 999 \
  --wandb_group wave3-huber-loss
```
Also run with `--huber-delta 0.1` and `--huber-delta 0.5`.

### TandemFoil Paper (EMA enabled)
```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil_paper \
  --optimizer lion --lr 1e-4 --cosine-t-max 10 \
  --grad-clip 1.0 --weight-decay 1e-2 --ema-decay 0.999 \
  --model-slices 64 --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --enable-fourier --enable-te-coord-frame --enable-cp-panel \
  --enable-cp-panel-tandem-only --asinh-pressure \
  --residual-prediction --enable-pressure-prior-addition \
  --loss-fn huber --huber-delta 1.0 \
  --epochs 999 \
  --wandb_group wave3-huber-loss
```

### AirfRANS (golden config + Huber loss)
```bash
cd target/icml2026 && python train.py \
  --dataset airfrans \
  --optimizer adamw --lr 6e-4 --cosine-t-max 10 \
  --grad-clip 1.0 --weight-decay 1e-2 --no-use-ema \
  --model-slices 64 --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --enable-fourier \
  --loss-fn huber --huber-delta 1.0 \
  --epochs 999 \
  --wandb_group wave3-huber-loss
```
Also run `--huber-delta 0.1`.

### DrivAerML (golden config + Huber loss)
```bash
SENPAI_MAX_EPOCHS=9999 python train.py \
  --dataset drivaerml \
  --optimizer adamw --lr 5e-4 --cosine-t-max 30 \
  --grad-clip 1.0 --weight-decay 1e-2 --no-use-ema \
  --model-slices 64 --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --enable-fourier \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --loss-fn huber --huber-delta 1.0 \
  --epochs 999 \
  --wandb_group wave3-huber-loss
```

## Current Baselines

| Dataset | Metric | Baseline | Best PR |
|---------|--------|----------|---------|
| TandemFoil | `val_primary/surface_pressure_mae` | 26.06 | #2887 |
| TandemFoil Paper | `val_primary/field_mse` | TBD | #2947/#2948/#2949 |
| AirfRANS | `val_primary/surface_mse` | 0.000627 | #2902 |
| DrivAerML | `val_primary/surface_rel_l2_pct` | 3.997% | #2898 |

## Notes

- Huber loss is a well-established regression robustification technique — no architectural changes needed, pure loss substitution
- If delta=1.0 does not beat MSE, try delta=0.1 (more aggressive outlier rejection)
- The PyTorch `F.huber_loss` implementation is correct and efficient — no custom CUDA needed
- For DrivAerML, the metric is already relative L2 — Huber may help if outlier surface points are causing MSE to over-weight hard regions

### CODE REVIEW

**Comment by morganmcg1** [comment]
Closing to narrow the queue for the final ICML benchmark sprint. Current priority is benchmark-specific optimization and benchmark-faithful reporting for DrivAerML, TandemFoil paper benchmark, AirfRANS full Surf/Vol, and the Tandem parity anchor. If this idea still matters later, we can reopen it in a benchmark-specific form.
